linchpin/__init__.py

```diff
@@ -547,7 +547,8 @@ def get_pf_data_from_rundb(self, targets, run_id=None, tx_id=None):
         return pinfile
 
 
-    def do_action(self, provision_data, action='up', run_id=None, tx_id=None):
+    def do_action(self, provision_data, action='up', run_id=None, tx_id=None,
+                  vault_pass=''):
```

Comment:
This functionally changes the LinchPin api when it is not needed. A better way to approach this would be to use the context and pass it in similarly to the way `outfile` is passed [here](https://github.com/CentOS-PaaS-SIG/linchpin/blob/develop/linchpin/shell/__init__.py#L145).